service.go

```diff
@@ -1704,39 +1711,28 @@ func (s *Service) sendPayloadStats(payload []byte, logMetric *LogMetric, isSucce
 		AccountID:                 accountID,
 		ValidatorID:               validatorID,
 		GetPayloadLatency:         latency,
+		SlotUID:                   slotUID,
 	}
 	var (
 		isRelayProxyWin bool
+		fallback        SlotStatsRecord
 	)
 	k := fmt.Sprintf("slot-%v-parentHash-%v", out.GetSlot(), out.GetParentHash())
 	v, ok := s.slotStats.Get(k)
 	if ok {
 		if records, success := v.([]SlotStatsRecord); success {
 			for i, record := range records {
-				isRelayProxyWin = record.HeaderDeliveredBlockHash == out.GetBlockHash()
-				if isRelayProxyWin || i == len(records)-1 {
-					statsRecord.HeaderReqID = record.HeaderReqID
-					statsRecord.HeaderReqReceivedAt = record.HeaderReqReceivedAt
-					statsRecord.HeaderReqDuration = record.HeaderReqDuration
-					statsRecord.HeaderReqDurationInMs = record.HeaderReqDurationInMs
-					statsRecord.HeaderMsIntoSlot = record.HeaderMsIntoSlot
-					statsRecord.HeaderMsIntoSlotWithDelay = record.HeaderMsIntoSlotWithDelay
-					statsRecord.HeaderDelayInMs = record.HeaderDelayInMs
-					statsRecord.HeaderMaxDelayInMs = record.HeaderMaxDelayInMs
-					statsRecord.HeaderSucceeded = record.HeaderSucceeded
-					statsRecord.HeaderDeliveredBlockHash = record.HeaderDeliveredBlockHash
-					statsRecord.HeaderBlockValue = record.HeaderBlockValue
-					statsRecord.HeaderUserAgent = record.HeaderUserAgent
-					statsRecord.PubKey = record.PubKey
-					statsRecord.GetHeaderLatency = record.GetHeaderLatency
-
-					// needed for stakely
-					statsRecord.AccountID = record.AccountID
-					accountID = record.AccountID
-					statsRecord.ValidatorID = record.ValidatorID
-					validatorID = record.ValidatorID
+				if i == len(records)-1 {
+					fallback = record
+				}
+				if record.HeaderDeliveredBlockHash == out.GetBlockHash() && record.SlotUID == statsRecord.SlotUID {
+					mergeSlotStats(record, statsRecord)
+					isRelayProxyWin = true
 					break
 				}
+				if !isRelayProxyWin {
```

Comment:
should this only occur on the last record? Right now we mergeSlotStats every time.